### ISSUE TITLE
obvious type is reported as unknown

### ISSUE THREAD

**Issue by ilya-klyuchnikov** [microsoft/pyright#3732]
**Describe the bug**

The type of the field is not inferred from the type definition.

**Screenshots or Code**

```python
# pyright: strict
from dataclasses import dataclass

@dataclass
class MyData:
    position: int

def example(d: MyData) -> int:
    match d:
        case MyData(x):
            return x
```

**Additional context**

```
pyright example2.py

example2.py:14:20 - error: Return type is unknown (reportUnknownVariableType)
example2.py:13:21 - error: Variable "x" is not accessed (reportUnusedVariable)
2 errors, 0 warnings, 0 informations 
```

I would expect that the inferred type for `x` is `int`

**Comment by erictraut** [comment]
Thanks for the bug report. This turns out to be the same underlying bug as https://github.com/microsoft/pyright/issues/3728. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.263, which I just published.
